internal/dag/cache.go

```diff
@@ -54,6 +54,11 @@ type Meta struct {
 func (kc *KubernetesCache) Insert(obj interface{}) bool {
 	switch obj := obj.(type) {
 	case *v1.Secret:
+		if _, hasCA := obj.Data["ca.crt"]; obj.Type != v1.SecretTypeTLS && !hasCA {
```

Comment:
We should verify we have this rule documented somewhere. I know `kubernetes.io/tls` secrets aren't even well documented, so I could see folks creating generic secrets with the proper key (e.g. `ca.crt`) and having them be ignored.

Reply:
>  so I could see folks creating generic secrets with the proper key (e.g. ca.crt) and having them be ignored.

I think the logic is the opposite. If I created a secret that had a key `ca.crt` then every time it was touched it would trigger a dag rebuild.

Reply:
OH right, I misread that.

Reply:
It is super subtle, and kind of annoying that we didn't also guard this with a Secret Type so we don't have to rely on the presence of the key.